src/mongoc/mongoc-cluster.c

```diff
@@ -1866,51 +1896,18 @@ mongoc_cluster_get_max_bson_obj_size (mongoc_cluster_t *cluster,
  *--------------------------------------------------------------------------
  */
 int32_t
-mongoc_cluster_get_max_msg_size (mongoc_cluster_t *cluster,
-                                 uint32_t         *server_id /* IN */)
+mongoc_cluster_get_max_msg_size (mongoc_cluster_t *cluster)
 {
-   mongoc_server_description_t *sd;
-   mongoc_cluster_node_t *node;
-   mongoc_topology_t *topology;
-   int32_t max_msg_size = -1;
-   bool single_threaded;
-
-   topology = cluster->client->topology;
-   mongoc_mutex_lock (&topology->mutex);
-   single_threaded = topology->single_threaded;
-   mongoc_mutex_unlock (&topology->mutex);
-
-   if (server_id) {
-      if (!single_threaded) {
-         /* if we can, use information stored in cluster node */
-         node = mongoc_set_get (cluster->nodes, *server_id);
-         if (node) {
-            max_msg_size = node->max_msg_size;
-         }
-      } else {
-         /* otherwise, check the server description */
-         mongoc_mutex_lock (&topology->mutex);
-         sd = mongoc_set_get (topology->description.servers, *server_id);
-         if (sd) {
-            max_msg_size = sd->max_msg_size;
-         }
-         mongoc_mutex_unlock (&topology->mutex);
-      }
-   }
+   int32_t max_msg_size = MONGOC_DEFAULT_MAX_MSG_SIZE;
 
-   if (max_msg_size == -1) {
-      max_msg_size = MONGOC_CLUSTER_DEFAULT_MAX_MSG_SIZE;
-
-      /* if no server was found or given, get min of all values */
-      if (!single_threaded) {
-         mongoc_set_for_each (cluster->nodes,
-                              _mongoc_cluster_min_of_max_msg_size_nodes,
-                              &max_msg_size);
-      } else {
-         mongoc_set_for_each (topology->description.servers,
-                              _mongoc_cluster_min_of_max_msg_size_sds,
-                              &max_msg_size);
-      }
+   if (!cluster->client->topology->single_threaded) {
+      mongoc_set_for_each (cluster->nodes,
+                           _mongoc_cluster_min_of_max_msg_size_nodes,
+                           &max_msg_size);
+   } else {
+      mongoc_set_for_each (cluster->client->topology->description.servers,
```

Comment:
actually, this only gets run in the single-threaded case, so I think it should be ok